astropy/cosmology/_io/mapping.py

```diff
@@ -134,16 +134,34 @@
      'cosmo_name': 'Planck18', ...
 """
 
+from __future__ import annotations
+
 import copy
-from collections.abc import Mapping
+from collections.abc import ItemsView, Mapping
+from typing import TYPE_CHECKING, Any, Literal, Protocol, TypeVar
 
 from astropy.cosmology.connect import convert_registry
 from astropy.cosmology.core import _COSMOLOGY_CLASSES, Cosmology
 
-__all__ = []  # nothing is publicly scoped
+if TYPE_CHECKING:
+    from astropy.cosmology._typing import _CosmoT
+
+    _MapT = TypeVar("_MapT", Mapping[str, Any])
+
+    _KT_co = TypeVar("_KT_co", covariant=True)  # Key type covariant containers.
+    _VT_co = TypeVar("_VT_co", covariant=True)  # Value type covariant containers.
+
+    class SupportsItems(Protocol[_KT_co, _VT_co]):
+        """Protocol for objects that support the ``items`` method."""
+
+        def items(self) -> ItemsView[_KT_co, _VT_co]:
+            ...
+
+
+__all__: list[str] = []  # nothing is publicly scoped
 
 
-def _rename_map(map, /, renames):
+def _rename_map(map: SupportsItems[str, Any], /, renames: dict[str, str]):
```

Comment:
First of all I don't see why this code couldn't just use `Mapping` and instead chooses to invent `SupportsItems`. Second, the return type annotation is missing. Third, the values in `map` should not be annotated with `Any`, they should be annotated with `TypeVar`.

Reply:
> First of all I don't see why this code couldn't just use Mapping and instead chooses to invent SupportsItems. 

Sure. `SupportsItems` is the more specific, but Mapping is fine.

> Second, the return type annotation is missing.

Fixed.

> Third, the values in map should not be annotated with Any, they should be annotated with TypeVar.

The mapping values can be multiple types, so `TypeVar` won't work here.